src/experimental/collection_new_iterable_model/shared/collectionInternalCommunication/server/impl/CollectionServerDispatcher.hpp

```diff
@@ -0,0 +1,240 @@
+                        typename __ServerType,                                                                                                                          /* NOLINT(bugprone-reserved-identifier) */
+                        typename __ServiceType,                                                                                                                         /* NOLINT(bugprone-reserved-identifier) */
+                        typename __ElementType,                                                                                                                         /* NOLINT(bugprone-reserved-identifier) */
+                        typename __AbstractConstIteratorType,                                                                                                           /* NOLINT(bugprone-reserved-identifier) */
+                        typename __ConstIteratorType,                                                                                                                   /* NOLINT(bugprone-reserved-identifier) */
+                        auto ( __ServiceType :: * __cbeginFunction )            ()                                                      const   -> __ConstIteratorType, /* NOLINT(bugprone-reserved-identifier) */
+                        auto ( __ServiceType :: * __cendFunction )              ()                                                      const   -> __ConstIteratorType, /* NOLINT(bugprone-reserved-identifier) */
+                        auto ( __ServiceType :: * __newAddressFunction )        ( __ElementType const *, bool * )                               -> __ElementType *,     /* NOLINT(bugprone-reserved-identifier) */
+                        auto ( __ServiceType :: * __removeConstFunction )       ( __AbstractConstIteratorType const & )                         -> bool,                /* NOLINT(bugprone-reserved-identifier) */
+                        auto ( __ServiceType :: * __removeConstArrayFunction )  ( __AbstractConstIteratorType const * const *, Size )           -> Size                 /* NOLINT(bugprone-reserved-identifier) */
+                > __CDS_OptimalInline auto __CollectionServerDispatcher <
+                        __ServerType,
+                        __ServiceType,
+                        __ElementType,
+                        __AbstractConstIteratorType,
+                        __ConstIteratorType,
+                        __cbeginFunction,
+                        __cendFunction,
+                        __newAddressFunction,
+                        __removeConstFunction,
+                        __removeConstArrayFunction
+                > :: __cbegin () const noexcept -> __AbstractDelegateIterator < __ElementType const > * {
+
+                    /* call the template received member function and wrap iterator in a new delegate */
+                    return Memory :: instance().create < __DelegateIterator < __ElementType const, __ConstIteratorType > > (
```

Comment:
__AbstractDelegateIterator typo

Reply:
not a typo, __DelegateIterator < Type, WrapperType > is derived from __AbstractDelegateIterator < Type >
reason : __DelegateIterator = adapter from standard __ConstIteratorType functions to __AbstractDelegateIterator functions 
i.e. __AbstractDelegateIterator requires .valid() member fn..  standard iterators do not provide .valid(),
therefore, __DelegateIterator will call appropriate function to call valid
Furthermore, Address / Node / HT / RBT Iterators should NOT be derived from __ADIterator, since that would reduce constexpr-ness level to cpp-17 and above for creation, and cpp-20 for destruction
so wrap around adapter type : __DelegateIterator, which IS derived from __ADIterator, containing the base Address / Node / HT/ RBT Iterator ( type given as template, while __ADIterator is not aware of base type, only of element type )